src/controller.py

```diff
@@ -17,6 +20,43 @@ def classify_img(self, img):
         result = {}
         return result
 
-    def make_clip(self, timestamp, path):
-        # using timestamp[0] (start time) and timestamp[1] (end time)
-        return path  # clip path
+    def make_clip(self, timestamp, path, outputPath=None):
+        # Args: timestamp:((int)t0, (int)t1), path: (string)"path/to/input/video.mp4"
+        #                                  outputPath: (string) "path/to/destination/video.mp4"
+
+        if isinstance(timestamp, type(None)) or isinstance(path, type(None)) \
+                                             or isinstance(timestamp[0], type(None)) \
+                                             or isinstance(timestamp[1], type(None)):
+            raise TypeError("Nonetype in arguments. Naughty naughty.")
+        if not os.path.isfile(path):
+            raise ValueError("No file at {}".format(path))
+        if isinstance(outputPath, type(None)):
+            clipPath = path[:-4] + "_subclip({},{})".format(timestamp[0], timestamp[1]) + path[-4:]
```

Comment:
`os.path.splitext(path)` would be more robust here than using `path[-4:]`.

Reply:
I'm not looking to get the file name, I'm extracting the file extension.

Reply:
I am trying to insert text right inbetween filename and .mp4 to turn filename.mp4 -> filename_moretext.mp4

Reply:
I believe that is what `splitext()` does.

Reply:
I am not a smart man.